plugins/modules/content_view_filter_rule.py

```diff
@@ -0,0 +1,335 @@
+        entity_opts=dict(scope=['content_view_filter']),
+    )
+
+    with module.api_connection():
+
+        # A filter always exists before we create a rule
+        # Get a reference to the content filter that owns the rule we want to manage
+        cv_scope = module.scope_for('content_view')
+        cvf_scope = module.scope_for('content_view_filter')
+        cvf = module.lookup_entity('content_view_filter')
+
+        # figure out what kind of filter we are working with
+        filter_type = cvf['type']
+        rule_spec = globals()['content_filter_rule_%s_spec' % (filter_type)]
+
+        # trying to find the existing rule is not simple...
+        search_scope = cvf_scope
+        content_view_filter_rule = None
+
+        if filter_type != 'erratum' and module.foreman_params['name'] is None:
+            module.fail_json(msg="The 'name' parameter is required when creating a filter rule for rpm, container, package_group or modulemd filters.")
+
+        if filter_type == 'erratum':
+            # this filter type supports many rules
+            # there are really 2 erratum filter types by_date and by_id
+            # however the table backing them is denormalized to support both, as is the api
+            # for an erratum filter rule == errata_by_date rule, there can be only one rule per filter. So that's easy, its the only one
+            if 'errata_id' in module.foreman_params:
+                # we need to search by errata_id, because it really doesn't have a name field.
+                rule_spec = content_filter_rule_erratum_id_spec
+                search_scope['errata_id'] = module.foreman_params['errata_id']
+            content_view_filter_rule = module.find_resource('content_view_filter_rules', None, params=search_scope, failsafe=True)
+
+        elif filter_type in ('rpm', 'docker'):
+            # these filter types support many rules
+            # the name is the key to finding the proper one and is required for these types
+            content_view_filter_rule = module.find_resource_by_name('content_view_filter_rules', module.foreman_params['name'],
+                                                                    params=search_scope, failsafe=True)
+
+        elif filter_type == 'package_group':
+            # this filter type support many rules
+            # the name is the key to finding the proper one and is required for these types
+            # uuid is also a required value creating, but is implementation specific and not easily knowable to the end user - we find it for them
+            content_view_filter_rule = module.find_resource_by_name('content_view_filter_rules', module.foreman_params['name'],
+                                                                    params=search_scope, failsafe=True)
+
+            package_group = module.find_resource_by_name('package_groups', module.foreman_params['name'], params=cv_scope)
+            module.foreman_params['uuid'] = package_group['uuid']
```

Comment:
I'm wondering if simplifying this to the following will make it better readable or not?

```suggestion
        elif filter_type in ('rpm', 'docker', 'package_group'):
            # these filter types support many rules
            # the name is the key to finding the proper one and is required for these types
            content_view_filter_rule = module.find_resource_by_name('content_view_filter_rules', module.foreman_params['name'],
                                                                    params=search_scope, failsafe=True)

            if filter_type == 'package_group':
            # uuid is also a required value creating, but is implementation specific and not easily knowable to the end user - we find it for them
                package_group = module.find_resource_by_name('package_groups', module.foreman_params['name'], params=cv_scope)
                module.foreman_params['uuid'] = package_group['uuid']
```

Reply:
Yes. That works and means less maintenance.